tests/test_hvault.rs

```diff
@@ -186,7 +216,255 @@ async fn test_hashivault_client_address_load() -> Result<(), anyhow::Error> {
+async fn test_hashivault_auth_approle_secret() -> Result<(), anyhow::Error> {
+    test_setup().await?;
+
+    let approle_mount = "approle-secret-path";
+    let role_with_secret = "with-secret";
+
+    let client = hashivault_test_client();
+    enable_auth(&client, approle_mount, "approle").await?;
+
+    vault_set_app_role(&client, approle_mount, role_with_secret, true).await?;
+    let role_id = vaultrs::auth::approle::role::read_id(&client, approle_mount, role_with_secret).await?;
+    let secret = vaultrs::auth::approle::role::secret::generate(&client, approle_mount, role_with_secret, None).await?;
+
+    env::set_var("VAULT_AUTH_SECRET_ID", secret.secret_id);
+
+    let auth = HashiVaultAuth::AppRole {
+        mount_path: approle_mount.to_string(),
+        role_id: role_id.role_id.to_string(),
+        secret_id_path: None,
+    };
+
+    let vault_context = create_dummy_auth_context(None, auth, None, None);
+    let actual_client = build_client(&vault_context).await.expect("Should login with no errors");
+
+    assert!(!actual_client.settings.token.is_empty());
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_hashivault_auth_approle_secret_path() -> Result<(), anyhow::Error> {
+    test_setup().await?;
+
+    let approle_mount = "approle-secret";
+    let role_with_secret = "with-secret";
+
+    let client = hashivault_test_client();
+    enable_auth(&client, approle_mount, "approle").await?;
+
+    vault_set_app_role(&client, approle_mount, role_with_secret, true).await?;
+    let role_id = vaultrs::auth::approle::role::read_id(&client, approle_mount, role_with_secret).await?;
+    let secret = vaultrs::auth::approle::role::secret::generate(&client, approle_mount, role_with_secret, None).await?;
+
+    let secret_id_path = env::temp_dir().join("secret-id");
+    fs::write(secret_id_path.clone(), secret.secret_id.to_string())?;
+
+    let auth = HashiVaultAuth::AppRole {
+        mount_path: approle_mount.to_string(),
+        role_id: role_id.role_id.to_string(),
+        secret_id_path: secret_id_path.to_string_lossy().to_string().into(),
+    };
+
+    let vault_context = create_dummy_auth_context(None, auth, None, None);
+    let actual_client = build_client(&vault_context).await.expect("Should login with no errors");
+
+    assert!(!actual_client.settings.token.is_empty());
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_hashivault_auth_jwt() -> Result<(), Error> {
+    test_setup().await?;
+
+    let jwt_mount = "jwt-env";
+    let role = "test-env";
+
+    let client = hashivault_test_client();
+    let jwt_alg = jwt::Algorithm::ES256;
+    let jwt_key = vault_configure_jwt(&client, jwt_mount).await?;
+
+    let token_claims = create_jwt_token();
+
+    vault_configure_jwt_role(&client, jwt_mount, role, &token_claims).await?;
+
+    let token = jwt::encode(
+        &jwt::Header::new(jwt_alg),
+        &token_claims,
+        &jwt_key.to_encoding_key(),
+    )?;
+
+    env::set_var("VAULT_AUTH_JWT_TOKEN", token);
```

Comment:
I can't make this test pass on my side, it seems this variable isn't used as `client.rs` will always look for file path locally. It seems you wanted the client to use this variable somehow?

Reply:
Yes, I fixed the code to use this constant if the environment variable doesn't exists.